Signal/src/ViewControllers/HomeView/HomeViewCell.m

```diff
@@ -111,15 +118,25 @@ - (void)commontInit
     self.snippetLabel.font = [self snippetFont];
     self.snippetLabel.numberOfLines = 1;
     self.snippetLabel.lineBreakMode = NSLineBreakByTruncatingTail;
-    [self.snippetLabel setContentHuggingHorizontalLow];
-    [self.snippetLabel setCompressionResistanceHorizontalLow];
 
-    UIStackView *bottomRowView = [[UIStackView alloc] initWithArrangedSubviews:@[
+    self.typingIndicatorView = [TypingIndicatorView new];
+
+    self.previewStackView = [[UIStackView alloc] initWithArrangedSubviews:@[
         self.snippetLabel,
+        self.typingIndicatorView,
+    ]];
+    self.previewStackView.axis = UILayoutConstraintAxisVertical;
+    self.previewStackView.alignment = UIStackViewAlignmentLeading;
+    [self.previewStackView setContentHuggingHorizontalLow];
+    [self.previewStackView setCompressionResistanceHorizontalLow];
+
+    UIStackView *bottomRowView = [[UIStackView alloc] initWithArrangedSubviews:@[
+        self.previewStackView,
         self.messageStatusView,
     ]];
+    
     bottomRowView.axis = UILayoutConstraintAxisHorizontal;
-    bottomRowView.alignment = UIStackViewAlignmentLastBaseline;
+    bottomRowView.alignment = UIStackViewAlignmentCenter;
```

Comment:
I know it's subtle point but we were intentionally aligning on baseline. It's a couple px different. 

To keep the existing behavior you might have to do something like add a [viewForLastBaseLineLayout](https://developer.apple.com/documentation/uikit/uiview/1622633-viewforlastbaselinelayout?language=objc) to the typingIndicatorView.

Reply:
Okay.  I ended up reworking this entirely to hide the snippet label by changing its text.  IMO that'll be simplest.